### ISSUE TITLE
Watir Logger

### ISSUE THREAD

**Issue by titusfortner** [watir/watir#548]
Similar to [what we did for Selenium](https://github.com/SeleniumHQ/selenium/pull/3474)

Would we need to / want to tie into the Selenium Logger, or have a completely separate implementation?

Only function for this right now would be to move over the `Kernel#warn` implementations.

**Comment by titusfortner** [comment]
@p0deje I'm trying to think of how it makes sense to combine this with Selenium Logger. Does it make sense to have multiple loggers? I'm mostly thinking that we would want everything piped to the same place without having to specify it multiple times. But then again, maybe we do want to allow different levels of logs to go to different places.

Would it make sense to subclass the Selenium class and update `#formatter` before taking `#info`, `#warn` actions? Or should we open up the Selenium code itself to accept a `:progname` parameter?

```
Watir.logger.warn("Usage of Element#foo deprecated")
# or
Selenium::WebDriver.logger.warn("Usage of Element#foo deprecated", progname: 'Watir')
```

The functionality I'm thinking would ideally allow any add-on test software to tie into it:
```
2017-07-21 13:35:27 WARN Watir Usage of Element#foo deprecated
2017-07-21 13:35:27 INFO Selenium -> POST session/3a07d909e670759791566264565ad382/element
2017-07-21 13:35:27 INFO Selenium    >>> http://127.0.0.1:9522/session/3a07d909e670759791566264565ad382/element | {"using":"id","value":"test1"}
2017-07-21 13:35:27 INFO Selenium <- {"sessionId":"3a07d909e670759791566264565ad382","status":0,"value":{"ELEMENT":"0.28780905327957207-1"}}
2017-07-21 13:35:27 INFO page-object Consider taking action x instead of y  
```

**Comment by p0deje** [comment]
I don't think it is a good idea to couple Watir and Selenium loggers anyhow - those are separate, so I might want to set Selenium logger to `:debug` level and redirect to `selenium.log` (to get all the messages + driver logs) while keeping Watir logger at `:warn` level and redirect to `watir.log`.

1 program = 1 log.

P.S. Sorry for getting back that late - I was on vacation with limited access to the internet.

**Comment by titusfortner** [comment]
Implemented: 4977d44d10767274f4af5de43dbd6418d164b594
